addons/slack/template/index.js

```diff
@@ -0,0 +1,116 @@
+const slack = require('./slack.js');
+const Sdk = require('@fusebit/add-on-sdk');
+const { WebClient } = require('@slack/web-api');
+const Superagent = require('superagent');
+
+module.exports = async (ctx) => {
+    const sdk = {
+        bot : new WebClient(ctx.configuration.slack_bot_access_token),
+        user: ctx.configuration.slack_user_access_token ? new WebClient(ctx.configuration.slack_user_access_token) : null
+    }
+
+    // Event registration/deregistration support
+    if(ctx.url.endsWith('events/registration')){
+        Sdk.debug("Received Events API registration request")
+        if(ctx.configuration.fusebit_events_registration &&
+            ctx.configuration.fusebit_events_token){
+            Sdk.debug("Events API configuration is present")
+
+            // TODO What prevents someone who finds out this URL to hit it and 
```

Comment:
Is there an issue here to be recorded around adding some kind of authentication option scheme to function execution, or at least allows functions themselves to require a valid credential?

Reply:
Absolutely, filed fivequarters/q5#726